### PR TITLE
feat!: add subaccount support for v2010 apis

### PR DESCRIPTION
**Pull request by shwetha-manvinkurke**
Add ability to pass in subaccount sid to the client that is used to populate the v2010 API paths

[JIRA](https://issues.corp.twilio.com/browse/DI-961)

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-go/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -104,16 +116,17 @@ import (
 )
 
 func main() {
-	accountSid := os.Getenv("TWILIO_ACCOUNT_SID")
+	username := os.Getenv("TWILIO_ACCOUNT_SID")
 	authToken := os.Getenv("TWILIO_AUTH_TOKEN")
 	phoneNumber := "AVAILABLE_TWILIO_PHONE_NUMBER"
+	accountSid := os.Getenv("TWILIO_SUBACCOUNT_SID")
```
Comment (childish-sambino) [inline README.md:122]: Should keep the naming similar to what we do in the docs: accountSid, authToken, and subaccountSid.

https://www.twilio.com/docs/iam/api/subaccounts?code-sample=code-send-a-message-from-a-subaccount&code-language=Node.js&code-sdk-version=3.x

File: client/base_client.go [hunk 2]
```diff
@@ -6,6 +6,7 @@ import (
 )
 
 type BaseClient interface {
+	GetAccountSid() string
```
Comment (childish-sambino) [inline client/base_client.go:9]: I don't think this needs to be part of the interface.

File: twilio/rest/accounts/v1/docs/DefaultApi.md [hunk 3]
```diff
@@ -53,8 +52,8 @@ Name | Type | Description
 
 ### HTTP request headers
 
-- **Content-Type**: application/x-www-form-urlencoded, 
-- **Accept**: application/json, 
+- **Content-Type**: application/x-www-form-urlencoded
```
Comment (childish-sambino) [inline twilio/rest/accounts/v1/docs/DefaultApi.md:55]: I thought these changes were part of a different PR?

File: client/client.go [hunk 4]
```diff
@@ -30,6 +30,7 @@ type Client struct {
 	BaseURL    string
 	Edge       string
 	Region     string
+	AccountSid string
```
Comment (childish-sambino) [inline client/client.go:33]: I'm not seeing where this `AccountSid` is being used in building the v2010 path.

File: client/base_client.go (hunk 2 above)
Comment (shwetha-manvinkurke) [reply client/base_client.go:9]: The client [here](https://github.com/twilio/twilio-go/blob/subaccount-support/twilio/rest/api/v2010/api_default.go#L26) is of the type `BaseClient` and the only way I could expose the AccountSid in the implementation of that baseclient was to have a getter in the interface.

File: twilio/rest/accounts/v1/docs/DefaultApi.md (hunk 3 above)
Comment (shwetha-manvinkurke) [reply twilio/rest/accounts/v1/docs/DefaultApi.md:55]: The twilio-go was not regenerated after those changes.

File: client/client.go (hunk 4 above)
Comment (shwetha-manvinkurke) [reply client/client.go:33]: It is used in api_default.go. Example [here](https://github.com/twilio/twilio-go/blob/subaccount-support/twilio/rest/api/v2010/api_default.go#L113)

File: twilio/rest/accounts/v1/docs/DefaultApi.md (hunk 3 above)
Comment (childish-sambino) [reply twilio/rest/accounts/v1/docs/DefaultApi.md:55]: @thinkingserious Can this PR not be merged? https://github.com/twilio/twilio-go/pull/67

File: client/client.go (hunk 4 above)
Comment (childish-sambino) [reply client/client.go:33]: I see it now. Lost in all the docs changes. Can follow up on the generator PR.

File: twilio/rest/accounts/v1/docs/DefaultApi.md (hunk 3 above)
Comment (thinkingserious) [reply twilio/rest/accounts/v1/docs/DefaultApi.md:55]: That PR was replaced with https://github.com/twilio/twilio-go/pull/72, once approved, will merge.

File: twilio/twilio.go [hunk 5]
```diff
@@ -97,9 +97,9 @@ type Meta struct {
 	URL             *string `json:"url"`
 }
 
-// NewClient provides an initialized Twilio client.
-func NewClient(accountSID string, authToken string) *Twilio {
-	credentials := &client.Credentials{AccountSID: accountSID, AuthToken: authToken}
+// NewClientWithAccountSid provides an initialized Twilio client with accountSid.
+func NewClientWithAccountSid(username string, password string, accountSid string) *Twilio {
```
Comment (childish-sambino) [inline twilio/twilio.go:101]: I've a feeling we'll be wanting more optional params when constructing a `Twilio`. Can this be changed to accept a `struct` of optional params?

Comment (shwetha-manvinkurke) [reply twilio/twilio.go:101]: Optional params such as edge and region? Any ideas on what to call this func?

**Review by childish-sambino — APPROVED** [review]

File: twilio/rest/api/v2010/docs/DefaultApi.md [hunk 6]
```diff
@@ -829,7 +829,7 @@ Send a message from the account used to make the request
 Name | Type | Description
 ------------- | ------------- | -------------
 **ctx** | **context.Context** | context for authentication, logging, cancellation, deadlines, tracing, etc.
-**AccountSid** | **string** | The SID of the [Account](https://www.twilio.com/docs/iam/api/account) that will create the resource.
+**PathAccountSid** | **string** | The SID of the [Account](https://www.twilio.com/docs/iam/api/account) that will create the resource.
```
Comment (childish-sambino) [inline twilio/rest/api/v2010/docs/DefaultApi.md:832]: Should be part of `Other Parameters` below.

**Review by childish-sambino — APPROVED** [review]

**Pull request merged**